plugins/auto-sizes/tests/improve-sizes-test.php

```diff
@@ -24,85 +24,198 @@ public static function set_up_before_class(): void {
 
-		$expected = '<img sizes="(max-width: 1280px) 100vw, 1280px" src="test.jpg" />';
-		$result   = auto_sizes_improve_image_sizes_attribute( $content, $parsed_block );
+		$result = apply_filters( 'the_content', $block_content );
 
 		$this->assertStringContainsString( 'sizes="(max-width: 1280px) 100vw, 1280px" ', $result );
 	}
 
 	/**
-	 * Test the function with default alignment (contentSize).
+	 * Data provider.
 	 *
-	 * @covers ::auto_sizes_improve_image_sizes_attribute
+	 * @return array<array<string>> The image sizes.
 	 */
-	public function test_default_alignment(): void {
-		$content      = $this->get_image_tag( self::$image_id );
-		$parsed_block = array(
-			'attrs' => array(
-				// No alignment specified.
+	public function data_image_sizes_for_wide_alignment(): array {
+		return array(
+			'Return wideSize 1280px instead of thumb size 150px'  => array(
+				'thumbnail',
+			),
+			'Return wideSize 1280px instead of medium size 300px'  => array(
+				'medium',
+			),
+			'Return wideSize 1280px instead of large size 1024px'  => array(
+				'large',
+			),
+			'Return wideSize 1280px instead of full size 1080px'  => array(
+				'full',
 			),
 		);
+	}
 
-		$expected = '<img sizes="(max-width: 620px) 100vw, 620px" src="test.jpg" />';
-		$result   = auto_sizes_improve_image_sizes_attribute( $content, $parsed_block );
+	/**
+	 * Test the cover block with wide alignment.
+	 */
+	public function test_cover_block_with_wide_alignment(): void {
+		$image_url     = wp_get_attachment_image_url( self::$image_id, 'full' );
+		$block_content = '<!-- wp:cover {"url":"' . $image_url . '","id":' . self::$image_id . ',"dimRatio":50,"align":"wide","style":{"color":{}}} -->
+		<div class="wp-block-cover alignwide"><span aria-hidden="true" class="wp-block-cover__background has-background-dim"></span><img class="wp-block-cover__image-background wp-image-' . self::$image_id . '" alt="" src="' . $image_url . '" data-object-fit="cover"/><div class="wp-block-cover__inner-container"><!-- wp:paragraph {"align":"center","fontSize":"large"} -->
+		<p class="has-text-align-center has-large-font-size"></p>
+		<!-- /wp:paragraph --></div></div>
+		<!-- /wp:cover -->';
+
+		$result = apply_filters( 'the_content', $block_content );
+		$this->assertStringContainsString( 'sizes="(max-width: 1280px) 100vw, 1280px" ', $result );
+	}
 
+	/**
+	 * Test the image block with different image sizes and default alignment (contentSize).
+	 *
+	 * @dataProvider data_image_sizes_for_default_alignment
+	 *
+	 * @param string $image_size Image size.
+	 * @param string $expected   Expected output.
+	 * @param string $is_resize  Whether resize or not.
+	 */
+	public function test_image_block_with_default_alignment( string $image_size, string $expected, string $is_resize = '' ): void {
```

Comment:
I'd pass a `bool` to `$is_resize`, rather than a string. Will need to update the data provider cases to pass `true` instead of 'yes'.